### PR TITLE
Update UID system to match officail and Implement Merger

### PR DESCRIPTION
**Pull request by hemagx**
Update Unique ID system to match official now it cover all cash items include stackable ones.
Implement ForceSerial option in item database to force serial for any item.
Implement Merge Client interface to merge stackable items with serial numbers ( check npc/other/item_merge.txt ).

### CODE REVIEW

**Comment by kyeme** [comment]
👍

**Comment by hemagx** [comment]
i found some stuff that i should have deleted before i commit and commit message need update, tomorrow morning i will do this ~

sorry ~

**Comment by QQfoolsorellina** [comment]
you forgot to update main.sql 
INSERT INTO `sql_updates` (`timestamp`) VALUES (1435860840); --2015-07-02--18-14.sql

**Comment by hemagx** [comment]
@QQfoolsorellina 

thanks to remind me, i will update main.sql and item_db.sql files as well !

**Comment by hemagx** [comment]
@MishimaHaruna 

Can you check the SQL update file ? it doesn't seem to work well

**Comment by hemagx** [comment]
@MishimaHaruna 
i fixed the SQL ~

File: sql-files/upgrades/2015-07-02--18-14.sql [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    PREPARE STMT FROM @q;
+    EXECUTE STMT;
+  END IF;
+
+END $$
+
+CREATE PROCEDURE alter_if_exists(my_table TINYTEXT, my_column TINYTEXT, my_command TINYTEXT, my_predicate TEXT)
+BEGIN
+  set @dbname = DATABASE();
+  IF EXISTS (
```
Comment (dastgir) [inline sql-files/upgrades/2015-07-02--18-14.sql]: IF NOT EXISTS

**Comment by QQfoolsorellina** [comment]
I got this error when trying to load 2015-07-02--18-14.sql , how to fix it ?

[Err] 1267 - Illegal mix of collations (utf8_general_ci,IMPLICIT) and (utf8_unicode_ci,IMPLICIT) for operation '='
[Err] CALL alter_if_not_exists('item_db', 'forceserial', 'ADD COLUMN', 'TINYINT(1) UNSIGNED DEFAULT NULL AFTER `bindonequip`') ;
[Msg] Finished - Unsuccessfully
--------------------------------------------------

**Comment by hemagx** [comment]
@QQfoolsorellina

Could you please report this as issue ? i'm not sure what causes it, it works fine on my servers.

**Comment by dastgir** [comment]
Neither I can reproduce

**Comment by Jedzkie** [comment]
What client supports this feature?

**Comment by hemagx** [comment]
@Jedzkie unknow, but i'm pretty sure it works with 2012+

**Pull request merged**